middle_end/simplify_primitives.ml

```diff
@@ -147,6 +147,7 @@ let primitive (p : Lambda.primitive) (args, approxs) expr dbg ~size_int
         | Ostype_unix -> S.const_bool_expr expr (Sys.os_type = "Unix")
         | Ostype_win32 -> S.const_bool_expr expr (Sys.os_type = "Win32")
         | Ostype_cygwin -> S.const_bool_expr expr (Sys.os_type = "Cygwin")
+        | Backend_type -> S.const_int_expr expr 0 (* tag 0 is the same as Native *)
```

Comment:
indeed, I did have it in mind, but forget about the middle end, will do that change, thanks